tests/request/request.builder.spec.js

```diff
@@ -1,54 +1,22 @@
 const nock = require('nock');
 const fs = require('fs');
 
-const { YotiRequest } = require('../../src/request/request');
-const { RequestBuilder } = require('../../');
+const { RequestBuilder, Payload } = require('../../');
 const yotiPackage = require('../../package.json');
 
 const PEM_FILE_PATH = './tests/sample-data/keys/node-sdk-test.pem';
 const PEM_STRING = fs.readFileSync(PEM_FILE_PATH, 'utf8');
 const API_BASE_URL = 'https://api.example.com';
 const API_ENDPOINT = '/some-endpoint';
-
-/**
- * Assert that the signed request was built correctly.
- *
- * @param {Request} request
- */
-const assertExpectedRequest = (request, done) => {
-  expect(request).toBeInstanceOf(YotiRequest);
-
-  // Check that auth headers are present.
-  request.execute()
-    .then((response) => {
-      const sentHeaders = response.getParsedResponse().headers;
-
-      const expectedHeaders = {
-        'X-Yoti-SDK': 'Node',
-        'X-Yoti-SDK-Version': `Node-${yotiPackage.version}`,
-        'Content-Type': 'application/json',
-        Accept: 'application/json',
-      };
-
-      Object.keys(expectedHeaders).forEach((header) => {
-        const sentHeader = sentHeaders[header.toLowerCase()];
-        expect(sentHeader).toBe(expectedHeaders[header], header);
-      });
-
-      const expectedMatchHeaders = {
-        'X-Yoti-Auth-Key': new RegExp('^[a-zA-Z0-9/+]{392}$'),
-        'X-Yoti-Auth-Digest': new RegExp('^[a-zA-Z0-9/+=]{344}$'),
-      };
-
-      Object.keys(expectedMatchHeaders).forEach((header) => {
-        const sentHeader = sentHeaders[header.toLowerCase()];
-        expect(sentHeader).toMatch(expectedMatchHeaders[header], header);
-      });
-
-      done();
-    })
-    .catch(done);
+const CONTENT_TYPE_HEADER_NAME = 'Content-Type';
+const CONTENT_TYPE_JSON = 'application/json';
+const DEFAULT_HEADERS = {
+  'X-Yoti-SDK': 'Node',
+  'X-Yoti-SDK-Version': `Node-${yotiPackage.version}`,
+  Accept: CONTENT_TYPE_JSON,
+  'X-Yoti-Auth-Digest': new RegExp('^[a-zA-Z0-9/+=]{344}$'),
```

Comment:
Can we easily use the `DIGEST_KEY_PATTERN = /^[a-zA-Z0-9/+=]{344}$/` value here, instead of redeclaring? No worries if not

Reply:
We could move all common test constants/data in to a separate file, but we'd want to do this across all tests. As discussed, we'll leave it as is for now.